frontend/address-book/src/components/Main.jsx

```diff
@@ -1,13 +1,87 @@
-import {currentSearchTerm} from "../states";
-import {useRecoilValue} from "recoil";
+import {currentSearchTerm, modalData, modalState} from "../states";
+import {useRecoilState, useRecoilValue} from "recoil";
+import {useState} from 'react';
 
 
 const Main = (props)=>{
     // will need to add keys here for firstName, lastName, address
-    const currentState = useRecoilValue(currentSearchTerm);
+    const currentSearchValue = useRecoilValue(currentSearchTerm);
+    const [modalCustomerData, setModalData] = useRecoilState(modalData);
+    const [modalVisibilityState, setModalState] = useRecoilState(modalState);
+    const [customerState, setState] = useState([{
+        _id: "C91G92CJ77YP8NW6T5VYG862",
+        firstName: "Lawrence ",
+        lastName: "Bennett",
+        street1: "1826 Elk Street",
+        city: "Irvine",
+        state: "California"
+    },{
+        _id: "ECS7OSQY0APQX30KV8MZVOAG",
+        firstName: "Ralph",
+        lastName: "Horstman",
+        street1: "3239 Lauren Drive",
+        city: "Madison",
+        state: "Wisconsin"
+    },{
+        _id: "HFJ949U32DGICT4ZOBDRBENV",
+        firstName: "Christopher",
+        lastName: "Thomas",
+        street1: "3282 Ritter Street",
+        city: "Lexington",
+        state: "Alabama"
+    },{
+        _id: "9JA0JZYH4G79RAHDQO4J0UR0",
+        firstName: "Connie",
+        lastName: "Turk",
+        street1: "3444 Upland Avenue",
+        city: "Holland",
+        state: "Ohio"
+    }])
+    //Customer state includes Name, Address, Hardcoded data for the time being until backend finishes API Routes
+    
     return (
         <>
-            <p>The current search is: {currentState}</p>
+            {customerState ? (
+                <div className="row" >
+                    {customerState.filter(customer => {
+                        if (!currentSearchValue){
+                            return customer;
+                        } else if (customer.Name.toLowerCase().includes(currentSearchValue.toLowerCase())) {
+                            return customer;
+                        }
+                    }).map(customer => (
+                        <div className="customerCard" id={customer._id} key={customer._id}>
+                            <div className="customerCardBody">
+                                <div className="customerCardName">{customer.firstName} {customer.lastName}</div>
```

Comment:
Want to test this, and/or see a demo prior to merging due to the single brace.

Reply:
[Here is a video showing off how it works](https://youtu.be/Jb-RXazq42c), I noticed that a subvariable was missed when refactoring for the modal support, so I also went and updated that as well.